Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -1,147 +1,323 @@
+  simpa only [Cardinal.toNat_lift] using congr_arg Cardinal.toNat
+    (sepDegree_eq_of_equiv F E E' i)
+
+@[simp]
+theorem sepDegree_self : sepDegree F F = 1 :=
+  le_antisymm
+    (show sepDegree F F ≤ 1 from Cardinal.le_one_iff_subsingleton.2 AlgHom.subsingleton)
+      (Cardinal.one_le_iff_ne_zero.2 (sepDegree_nezero F F).out)
+
+@[simp]
+theorem finSepDegree_self : finSepDegree F F = 1 := by
+  simp only [finSepDegree, sepDegree_self, Cardinal.one_toNat]
+
+@[simp]
+theorem sepDegree_bot : sepDegree F (⊥ : IntermediateField F E) = 1 := by
+  have := sepDegree_eq_of_equiv _ _ _ (IntermediateField.botEquiv F E)
+  rwa [sepDegree_self, Cardinal.lift_one, ← Cardinal.lift_one.{u, v}, Cardinal.lift_inj] at this
+
+@[simp]
+theorem finSepDegree_bot : finSepDegree F (⊥ : IntermediateField F E) = 1 := by
+  simp only [finSepDegree, sepDegree_bot, Cardinal.one_toNat]
+
+@[simp]
+theorem sepDegree_top : sepDegree F (⊤ : IntermediateField F E) = sepDegree F E := by
+  simpa only [Cardinal.lift_id] using sepDegree_eq_of_equiv F _ E IntermediateField.topEquiv
+
+@[simp]
+theorem finSepDegree_top : finSepDegree F (⊤ : IntermediateField F E) = finSepDegree F E := by
+  simp only [finSepDegree, sepDegree_top]
+
+theorem sepDegree_bot' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : Cardinal.lift.{v} (sepDegree F (⊥ : IntermediateField E K)) =
+      Cardinal.lift.{w} (sepDegree F E) := by
+  exact sepDegree_eq_of_equiv _ _ _ ((IntermediateField.botEquiv E K).restrictScalars F)
+
+@[simp]
+theorem sepDegree_bot'' (K : Type v) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : sepDegree F (⊥ : IntermediateField E K) = sepDegree F E := by
+  simpa only [Cardinal.lift_id] using sepDegree_bot' F E K
+
+@[simp]
+theorem finSepDegree_bot' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : finSepDegree F (⊥ : IntermediateField E K) = finSepDegree F E := by
+  simpa only [Cardinal.toNat_lift] using congr_arg Cardinal.toNat (sepDegree_bot' F E K)
+
+@[simp]
+theorem sepDegree_top' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : sepDegree F (⊤ : IntermediateField E K) = sepDegree F K := by
+  simpa only [Cardinal.lift_id] using sepDegree_eq_of_equiv _ _ _
+    ((IntermediateField.topEquiv (F := E) (E := K)).restrictScalars F)
+
+@[simp]
+theorem finSepDegree_top' (K : Type w) [Field K] [Algebra F K] [Algebra E K]
+    [IsScalarTower F E K] : finSepDegree F (⊤ : IntermediateField E K) = finSepDegree F K := by
+  simp only [finSepDegree, sepDegree_top']
+
+/-- The standard isomorphism between `E →ₐ[F] K` and `E →ₐ[F] (normalClosure F E K)`. -/
+def algHomEquivNormalClosure (K : Type w) [Field K] [Algebra F K] :
+    (E →ₐ[F] K) ≃ (E →ₐ[F] (normalClosure F E K)) :=
+  ⟨fun f ↦ (IntermediateField.inclusion f.fieldRange_le_normalClosure) |>.comp
+    (AlgEquiv.ofInjectiveField f).toAlgHom, normalClosure F E K |>.val.comp,
+      fun _ ↦ rfl, fun _ ↦ rfl⟩
```

Comment:
This is already merged as [normalClosure.algHomEquiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/NormalClosure.html#normalClosure.algHomEquiv)